Add test command

arturdm opened the pull request.

arturdm committed:
Add test command

WezSieTato commented on lib/src/command/test_command.dart:
```diff
@@ -0,0 +1,66 @@
+class TestCommand extends Command<Null> {
+  String get workingDirectory => _fromGlobalResults("working-directory");
+
+  String get projectName => _fromGlobalResults("project");
```
This code and above are duplicated. What do you think about create `MoronepoResults` class that will have these properties and `MoronepoCommand` that will has `moronepoResults` property and other commands will inherit it?

WezSieTato commented on test/command/run_command_integration_test.dart:
```diff
@@ -5,14 +5,15 @@ import 'package:test/test.dart';
     test("should notify if no projects found", () async {
-      final emptyDirectory = "${Directory.current.path}/test/command/test_project/empty_directory";
+      final emptyDirectory =
+          "${Directory.current.path}/test_resources/command/test_project/empty_directory";
```
Do we need `command` directory? Can we  move `test_project` and `test_project_fo_test` one directory up?

arturdm replied:
Currently, there's a 1-1 relation of the directory layout in tests and production code if those tests are in `command` directory.

WezSieTato commented on analysis_options.yaml:
```diff
@@ -1,6 +1,7 @@
 include: package:pedantic/analysis_options.1.8.0.yaml
 
 analyzer:
+  exclude: [test_resources/**]
```
Why we don't want to have beautiful test projects :P ?

arturdm replied:
I'll try to check how I can fix the analysis error that occurs there. It seemed like something related to imports but the project actually works if you try to run it from its directory :/.

arturdm committed:
Analyze test_resources too

This requires generating .packages file in flutter projects
by running pub get (added to .travis.yml)

arturdm committed:
Extract common command code to MoronepoCommand

arturdm committed:
Fix failing analysis

WezSieTato commented on test/current_directory.dart:
```diff
@@ -0,0 +1,10 @@
+
+import 'package:path/path.dart';
+
+final Directory testDirectory = Directory(join(
```
Filename is weird/misleading for me. What do you think about `test_directory`\`test_directories`?

arturdm replied:
I'll call it directories since `projectDirectory` is the one used and derived from `testDirectory`
